File: 1208-get-equal-substrings-within-budget/main.go
package main

func equalSubstring(s string, t string, maxCost int) int {

	var maxLen int = 0
	var cost int = 0
	var left int = 0
	var right int = 0

	for right < len(s) {
		cost += abs(int(s[right]) - int(t[right]))
		for cost > maxCost {
			cost -= abs(int(s[left]) - int(t[left]))
			left++
		}
		maxLen = max(maxLen, right-left+1)
		right++
	}
	return maxLen

}

func abs(a int) int {
	if a < 0 {
		return -a
	}
	return a
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

File: 1208-get-equal-substrings-within-budget/main.rs
impl Solution {
    pub fn equal_substring(s: String, t: String, max_cost: i32) -> i32 {
        
        let mut max_len = 0;
        let mut cost = 0;
        let mut left = 0;
        let mut right = 0;

        let s = s.as_bytes();
        let t = t.as_bytes();

        while right < s.len() {
            cost += (s[right] as i32 - t[right] as i32).abs();
            right += 1;

            while cost > max_cost {
                cost -= (s[left] as i32 - t[left] as i32).abs();
                left += 1;
            }

            max_len = max_len.max(right - left);
        }

        max_len as i32
    }


}